packages/addon-dev/sample-typescript-rollup.config.js

```diff
@@ -0,0 +1,113 @@
+/**
+  *  This file should live at ./config/rollup.config.js
+  *  and the babel config at ./config/babel.config.js (if the babel config is ESM)
+  *    _if the babel config is CJS, it can be at the addon root_
+  *
+  *  The command to build:
+  *
+  *    `rollup -c ./config/rollup.config.js`
+  */
+import path from 'path';
+
+import alias from '@rollup/plugin-alias';
+import multiInput from 'rollup-plugin-multi-input';
+import ts from 'rollup-plugin-ts';
+import { defineConfig } from 'rollup';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import { Addon } from '@embroider/addon-dev/rollup';
+
+import babelConfig from './babel.config';
+import packageJson from '../package.json';
+
+const addon = new Addon({
+  srcDir: 'src',
+  destDir: 'dist',
+});
+
+const transpilation = [
+  // Instruct rollup how to resolve ts and hbs imports
+  // (importing a template-only component, for example)
+  nodeResolve({ resolveOnly: ['./'], extensions: ['.js', '.ts', '.hbs'] }),
+
+  // Allow top-level imports (what folks are used to from v1 addons)
+  // During the build, anything referencing a top-level import will be
+  // replaced with a relative import.
+  // DANGER: it's somewhat easy to cause circular references with this tool
+  alias({
+    entries: [
+      {
+        find: '#types',
+        replacement: path.resolve('src', '-private', 'types.ts'),
+      },
+      {
+        find: packageJson.name,
+        replacement: path.resolve('src'),
+      },
+      {
+        find: `${packageJson.name}/(.*)`,
+        replacement: path.resolve('src/$1'),
+      },
+    ],
+  }),
+
+  // This babel config should *not* apply presets or compile away ES modules.
+  // It exists only to provide development niceties for you, like automatic
+  // template colocation.
+  // See `babel.config.json` for the actual Babel configuration!
+  ts({
+    // can be changed to swc or other transpilers later
+    // but we need the ember plugins converted first
+    // (template compilation and co-location)
+    transpiler: 'babel',
+    browserslist: false,
+    babelConfig,
```

Comment:
Why do we have to specify this? Shouldn't it resolve the babel.config.js automatically?

Reply:
it does, but hard errors if the babel config is ESM.
I opened an issue here: https://github.com/wessberg/rollup-plugin-ts/issues/167

Reply:
it turns out, because we can't use `"type": "module"` in our packages, the babel config needs to be cjs anyway